src/components/ProjectVersion/index.jsx

```diff
@@ -1,19 +1,178 @@
+  const dateFilter = () => {
+    if (beforeDate) {
+      return allDates.filter(
+        dateString => new Date(dateString) > new Date(beforeDate),
+      );
+    }
+
+    return false;
+  };
+
+  const updateVersions = (prevVersion, nextVersion) => {
+    usePageStatusStore.getState().setStatus({
+      beforeVersion: prevVersion,
+      afterVersion: nextVersion,
+    });
+  };
+
+  const onClickBackToProject = ev => {
+    ev.preventDefault();
+
+    usePageStatusStore.getState().clearProject();
+
+    navigate(-1);
+  };
+
+  const onClickSubmitButton = async ev => {
+    ev.preventDefault();
+    setIsLoaded(false);
+    const result = { beforeVersion, afterVersion };
+
+    updateVersions(result);
+
+    const pageList = await getPages(result);
+
+    usePageListStore.getState().setPages(pageList);
+
+    setIsLoaded(true);
+    navigate("/page");
+  };
+
+  const prevData = {
+    label: "이전 버전",
+    description: "지정한 버전 명이 없으면 시간으로 보여요!",
+    selects: [
+      {
+        id: "beforeDate",
+        onChange: onChangeBeforeDate,
+        options:
+          allDates &&
+          allDates.map(date => (
+            <option key={date} value={date}>
+              {date}
+            </option>
+          )),
+      },
+      {
+        id: "beforeVersion",
+        onChange: onChangeBeforeVersion,
+        options:
+          beforeDate &&
+          Object.entries(byDates[beforeDate]).map(([key, value]) => (
+            <option key={key} value={key}>
+              {convertLabel(value.label)}
+            </option>
+          )),
+      },
+    ],
+  };
+
+  const nextData = {
+    label: "이후 버전",
+    description: "지정한 버전 명이 없으면 시간으로 보여요!",
+    selects: [
+      {
+        id: "afterDate",
+        onChange: onChangeAfterDate,
+        options:
+          dateFilter() &&
+          dateFilter().map(date => (
+            <option key={date} value={date}>
+              {date}
+            </option>
+          )),
+      },
+      {
+        id: "afterVersion",
+        onChange: onChangeAfterVersion,
+        options:
+          afterDate &&
+          Object.entries(byDates[afterDate]).map(([key, value]) => (
+            <option key={key} value={key}>
+              {convertLabel(value.label)}
+            </option>
+          )),
+      },
+    ],
+  };
+
+  const contents = {
+    title: {
+      step: "02",
+      firstSentence: "비교할 해당 피그마 파일의",
+      secondSentence: "이전 / 최신 버전을 입력해 주세요",
+    },
+    buttons: [
+      { text: "이전", usingCase: "solid", handleClick: onClickBackToProject },
+      { text: "다음", usingCase: "solid", handleClick: onClickSubmitButton },
+    ],
+  };
+
   return (
     <>
+      {!isLoaded && (
+        <Modal>
+          <Loading />
+        </Modal>
+      )}
       <ContentsWrapper>
-        <Title />
-        <Select />
-        <Select />
+        <Title title={contents.title} />
+        <HorizontalAlign>
+          <Select selectInfo={prevData} />
+          <Select selectInfo={nextData} />
```

Comment:
P1: 이전 버전/ 다음 버전에 대한 데이터가 들어간다는 prop명이 명시적으로 들어가면 좋을 것 같아요.
Data는 일반적인 대명사라 직관적이지 않을 듯 합니다 😂

```suggestion
          <Select selectInfo={beforeVersion} />
          <Select selectInfo={afterVersion} />
```

Reply:
아래와 같이 변수명 수정했습니다.
prevData => prevVersionForm
nextData => nextVersionForm